coordinator/sources/sources.go

```diff
@@ -195,26 +195,83 @@ func (s *SourcesManager) Compile(
 		profilingOrDebugging,
 	)
 
-	cmd = exec.Command(
-		"bash",
-		filepath.Join(sourcesDir(), "scripts", "configure.sh"),
-	)
-	cmd.Dir = sourcesDir()
-	env := os.Environ()
-	if !profilingOrDebugging {
-		env = append(env, "BUILD_RELEASE=1")
+
+    scriptsDir := filepath.Join(sourcesDir(), "scripts")
+	if _, err := os.Stat(filepath.Join(scriptsDir, "install-build-tools.sh"); err == nil {
+		cmd = exec.Command(
+			"bash",
+			filepath.Join(scriptsDir, "install-build-tools.sh"),
+		)
+
+		cmd.Dir = sourcesDir()
+		env := os.Environ()
+		if !profilingOrDebugging {
+			env = append(env, "BUILD_RELEASE=1")
+		}
+		cmd.Env = env
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("Build-environment setup failed: %v\n\n%v", err, string(out))
+		}
+
+		logging.Infof(
+			"[Compile %s-%t]: Build-environment setup complete",
+			hash,
+			profilingOrDebugging,
+		)
 	}
-	cmd.Env = env
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("Configure failed: %v\n\n%v", err, string(out))
+
+	if _, err := os.Stat(filepath.Join(scriptsDir, "setup-dependencies.sh"); err == nil {
+		cmd = exec.Command(
+			"bash",
+			filepath.Join(sourcesDir(), "scripts", "setup-dependencies.sh"),
+		)
+		cmd.Dir = sourcesDir()
+		env := os.Environ()
+		if !profilingOrDebugging {
+			env = append(env, "BUILD_RELEASE=1")
+		}
+		cmd.Env = env
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("Dependency installation failed: %v\n\n%v", err, string(out))
+		}
+
+		logging.Infof(
+			"[Compile %s-%t]: Dependency installation complete",
+			hash,
+			profilingOrDebugging,
+		)
 	}
 
-	logging.Infof(
-		"[Compile %s-%t]: Configure script complete",
-		hash,
-		profilingOrDebugging,
-	)
+	if _, err := os.Stat(filepath.Join(scriptsDir, "configure.sh"); err == nil {
```

Comment:
Not a nit, and a reasonable question. ☺ You're correct that there shouldn't be a time (now) when both the legacy and modern scripts are present (save for a broken worktree, I guess). My only reason for preferring this style is actually just because it would further widen the margin on the left for no huge benefit. If we want to guarantee that only one of the two sets of scripts can be run, I'm happy to make that change.

Reply:
Well having both scripts run should not break anything, at least using the most recent version of configure.sh that was published in opencbdc-tx, but the configuration step will take twice as long. I can't really think of another reason why having them both be run would be problematic in the test controller -- unless someone were to commit a script named configure.sh that was explicitly not meant to be run in the test controller.

So basically, I don't think it really matters either way lol. 

Bonus reading for those interested:
If I were to get really nit-picky, I would say that there should be one path which must run both `install-build-tools.sh` and `setup-dependencies.sh`, and the other path would run `configure.sh`, such as what is written below:

```
if(exists(install-build-tools.sh) and exists(setup-dependencies.sh)) {
    run install-build-tools.sh
    run setup-dependencies.sh
}
else if(exists(configure.sh)) {
    run configure.sh
}
else {
    fail because dependencies not installed
}
```

But this sacrifices some readability, and a commit not containing both install-build-tools and setup-dependencies (or at least configure) should really never happen unless significant changes occur